x-pack/plugins/security/public/plugin.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Plugin, CoreSetup } from 'src/core/public';
+import { AnonymousPaths } from './anonymous_paths';
+import {
+  SessionExpired,
+  SessionTimeout,
+  SessionTimeoutInterceptor,
+  UnauthorizedResponseInterceptor,
+} from './session';
+
+export class SecurityPlugin implements Plugin<SecurityPluginSetup, SecurityPluginStart> {
+  public setup(core: CoreSetup, deps: {}) {
+    const { http, notifications } = core;
+    const { basePath } = http;
+    const anonymousPaths = new AnonymousPaths(basePath, [
+      '/login',
+      '/logout',
+      '/logged_out',
+      '/status',
```

Comment:
I'd much prefer to not have `/status` hard-coded here, but OSS can't call `security.anonymousPaths.register`...

Reply:
By the way, core (server side at least) has [a notion of authentication](https://github.com/elastic/kibana/blob/master/docs/development/core/server/kibana-plugin-server.routeconfigoptions.authrequired.md), why can't `anonymousePaths` be exposed by the core instead?

Reply:
Moving anonymous paths to the NP is turning into a bit of work. before I continue down this path, @elastic/kibana-platform do you mind confirming whether or not it makes sense to move this to the NP? Also, do you think it makes sense for it to be it's own "top-level service" or integrated within one of the existing services?

Reply:
I think it would make sense to be on the frontend's HTTP service. The data flow would probably be something like:
- Core's backend HTTP service exposes a readonly Set of paths
- the `ui_render_mixin` in the legacy platform serializes this set into the injected vars provided to the New Platform. We something similar already with plugins [here](https://github.com/joshdover/kibana/blob/b22cd3a6aef307ab1e2f383e2e1a2b4ccacf2579/src/legacy/ui/ui_render/ui_render_mixin.js#L223).
- Core's frontend InjectedMetadataService would expose this list
- Core's frontend HTTPService would read from InjectedMetadataService and expose to plugins (InjectedMetadataService is not exposed to plugins on purpose).

Once the ui_render_mixin is moved to the New Platform we could eliminate some of the threading here, but until then this probably makes the most sense.

Reply:
Technically, these are "frontend paths" where we ignore all 401s. For example the login page which is available at `/login`, the front-end http client ignores all 401s whenever the user is on this page. This makes me think that plugins should be registering their anonymous paths with the frontend http service during the `setup` lifecycle.

Reply:
Ah I understand now.

I think since this behavior (handling of 401s) is only used by Security (I think?) then wouldn't it be fine to keep this in the Security plugin and have the Security plugin register any OSS-owned paths itself? This is similar to what we do with the `features` plugin for registering features.

Reply:
This is essentially replacing this LP AngularJS service: https://github.com/elastic/kibana/blob/7ac8e4d9ccc2a9e064d8a5621cd62453d1006164/x-pack/legacy/plugins/xpack_main/public/services/path.js. The LP service has a hard-coded list of paths. This means that plugins themselves can't register their own paths, and instead this list must be modified. We've primarily addressed this in the NP version by allowing paths to be registered; however, the `/status` page is one of these unauthenticated paths. If we keep this NP anonymousPaths service in security, we can't have whatever OSS plugin that provides the status page register the `/status` route as being anonymous because that'd require OSS to have dependency on an x-pack plugin. This is the primarily reason we're trying to move this service to OSS In the NP.

It's also relevant that security isn't the only plugin which consumes the LP AngularJS service. It's also consumed by telemetry and reporting. 

Since the NP core already introduces the concept of "auth" via facilities like `core.http.registerAuth`, this felt like further validation that moving this to OSS would be appropriate.

Reply:
If it's used by other plugins, then I think moving it to the OSS/Core HttpService on the client side makes sense then 👍.

Reply:
Thanks @joshdover!